agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/diagnostics/status/RpAttachType.java

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.applicationinsights.agent.internal.diagnostics.status;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+// TODO (heya) to be moved to azure-monitor-opentelemetry-exporter so that it can be used in
+// AttachStatsbeat
+public enum RpAttachType {
+  AUTO,
+  MANUAL;
+
+  private static volatile RpAttachType attachType;
+
+  public static void setRpAttachType(Path agentPath, String markerFile) {
+    if (Files.exists(agentPath.resolveSibling(markerFile))) {
+      attachType = RpAttachType.AUTO;
+    } else {
+      attachType = RpAttachType.MANUAL;
+    }
+  }
```

Comment:
I think I'd leave this method as private in DiagnosticHelper, and have simple setter here (also because of moving this code to exporter where "agent" code isn't needed)